---
title: "pytorch-bert-crf-ner 에 관한 기록 01"
date: 2020-11-14 15:49:00 -0400
categories: python kobert bert
---

    RuntimeError: Input, output and indices must be on the current device site:stackoverflow.com

에러가 나는 경우

app.py를 수정해주니까 해결이 됐다.

     checkpoint = torch.load("./experiments/base_model_with_crf/best-epoch-16-step-1500-acc-0.993.bin",
                            map_location=torch.device('cpu'))
                            

이 부분을

다음과 같이 고친다

    checkpoint = torch.load("experiments/base_model_with_crf/best-epoch-16-step-1500-acc-0.993.bin",
                            map_location=torch.device('cuda:0')
                            )
                            
                            

app.py 실행 후 post 결과

    tensor([[   2,  681,  329,  294, 6116,  680,  342,  290, 6079, 3415, 7794,  969,
         3883,  522, 5561, 7582, 1261, 7078, 2472, 7794,  862, 5770, 3222, 7227,
         5859, 1185, 6042, 2872, 3863, 2705,   40,    3]], device='cuda:0')
    tensor([[   2,  681,  329,  294, 6116,  680,  342,  290, 6079, 3415, 7794,  969,
         3883,  522, 5561, 7582, 1261, 7078, 2472, 7794,  862, 5770, 3222, 7227,
         5859, 1185, 6042, 2872, 3863, 2705,   40,    3]], device='cuda:0') None tensor([[1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
         1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.]],
       device='cuda:0')
    len: 32, input_token:['[CLS]', '▁P', 'O', 'H', '를', '▁O', 'R', 'G', '로', '▁예측', '하는', '▁경우가', '▁있음', '▁(', '기', '타를', '▁기관', '으로', '▁분류', '하는', '▁거', '니', '▁어느', '정', '도', '▁그', '럴', '▁수', '▁있다고', '▁생각', ')', '[SEP]']
    len: 32, pred_ner_tag:['[CLS]', 'B-POH', 'I-POH', 'I-POH', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', '[SEP]']


잘 뜬다.




[jekyll-docs]: https://jekyllrb.com/docs/home
[jekyll-gh]:   https://github.com/jekyll/jekyll
[jekyll-talk]: https://talk.jekyllrb.com/
